assets/js/blocks/product-gallery/frontend.tsx

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import { store as interactivityApiStore } from '@woocommerce/interactivity';
+
+interface State {
+	[ key: string ]: unknown;
+}
+
+interface Context {
+	productGallery: { numberOfThumbnails: number };
+}
+
+interface Selectors {
+	productGallery: {
+		getNumberOfPages: ( store: unknown ) => number;
+	};
+}
+
+interface Store {
+	state: State;
+	context: Context;
+	selectors: Selectors;
+	ref: HTMLElement;
+}
+
+type SelectorsStore = Pick< Store, 'context' | 'selectors' >;
+
+interactivityApiStore( {
+	selectors: {
+		productGallery: {
```

Comment:
We should use `woocommerce` instead of `ProductGallery` as namespace. In according with this discussion, the namespace should match the plugin slug https://github.com/WordPress/gutenberg/discussions/53586. For the `ProductButton` we are using woocommerce.

Reply:
@gigitux I was reading the new Interactivity API proposal and it seems the namespace is currently not supported yet. Currently, we set the "namespace" as a property in the Interactivity API store, so basically to interact with the store  need to access each value using the `woocommerce` word:
```
actions.woocommerce.someAction...
selectors.woocommerce.someSelector...
```

However, in the future if the proposal is implemented, we won't need to use the `woocommerce` keyword since it will be set inside the `block.json` file and we can pass the `namespace` when instantiating the store:
```
// block.json
{
  ...
  "supports": {
    "interactivity": {
      "namespace": "woocommerce"
    "
  }

...
// view.ts or frontend.tsx file
store("woocommerce", {
  actions: {
    someAction: () => {}
  }
  selectors: {
    someSelector: () => {}
  } 

})

```

If the above is true, then if we use the `woocommerce` word now, in the future we would need to refactor all the files (that are using the Interactivity API store) to remove it, since it will not be necessary. Not sure if I'm missing something, but should we go with this approach?

Reply:
Yes, you got it! I linked that discussion to show you why we should use `woocommerce` instead of another value. 

> we go with this approach?

I think so. Since that it is a proposal, this could be refused or it could require time. I think that for now, we should we set the "namespace" as a property in the Interactivity API store.